shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java

```diff
@@ -279,13 +279,13 @@ private String getAlgorithmExpression(final TableRule tableRule, final boolean d
     }
     
     private String getShardingColumn(final ShardingStrategyConfiguration shardingStrategyConfig) {
+        String shardingColumn = this.defaultShardingColumn;
         if (shardingStrategyConfig instanceof ComplexShardingStrategyConfiguration) {
-            return ((ComplexShardingStrategyConfiguration) shardingStrategyConfig).getShardingColumns();
+            shardingColumn = ((ComplexShardingStrategyConfiguration) shardingStrategyConfig).getShardingColumns();
+        } else if (shardingStrategyConfig instanceof StandardShardingStrategyConfiguration) {
```

Comment:
Don't use else if here.

Reply:
why? There's only one condition that can be true, right?

Reply:
ok you are right.